File: my-folder/0057-insert-interval/solution.java
class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        List<int[]> list = new ArrayList<>();
        boolean inserted = false;
        for (int[] interval : intervals) {
            if (interval[1] < newInterval[0]) {
                list.add(interval);
            }
            else if (interval[0] > newInterval[1]) {
                if (inserted == false) {
                    list.add(newInterval);
                    inserted = true;
                }
                list.add(interval);
            }
            else {
                newInterval[0] = Math.min(newInterval[0], interval[0]);
                newInterval[1] = Math.max(newInterval[1], interval[1]);
            }
        }

        if (inserted == false) {
            list.add(newInterval);
        }
        int[][]res = new int[list.size()][2];
        for (int i = 0; i < list.size(); i++) {
            int[] arr = list.get(i);
            res[i] = new int[]{arr[0], arr[1]};
        }
        return res;
    }
}

File: my-folder/0057-insert-interval/solution.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        if not intervals: return [newInterval]

        res = []

        inserted = False

        for interval in intervals:
            if interval[1] < newInterval[0]:
                res.append(interval)
            elif interval[0] > newInterval[1]:
                if not inserted:
                    res.append(newInterval)
                    inserted = True
                res.append(interval)
            else:
                newInterval[0] = min(newInterval[0], interval[0])
                newInterval[1] = max(newInterval[1], interval[1])
        
        if not inserted:
            res.append(newInterval)

        return res
